app/models/document.rb

```diff
@@ -40,7 +40,7 @@ class Document < ApplicationRecord
   has_many :document_collections, through: :document_collection_groups
   has_many :features, inverse_of: :document, dependent: :destroy
 
-  before_save { check_if_locked_document(document: self) unless locked_changed? }
+  before_save { check_if_locked_document(document: self) unless locked_changed? || !changed? }
```

Comment:
Rather than changing this - which increases complexity in this callback could we add a conditional in the controller: https://github.com/alphagov/whitehall/blob/master/app/controllers/admin/export/document_controller.rb#L22 e..g `unless document.locked?`

Reply:
I'm not sure how we missed that 😊